rust/src/main.rs

```diff
@@ -76,37 +69,147 @@ fn setup_logging() -> Result<(), fern::InitError> {
         }).level(level_filter)
         .chain(io::stderr());
 
-    let path_result = path_for_log_file(XI_LOG_FILE);
-    // If the logging_file_path returned successfully, add the logfile capability to fern
-    if let Ok(logging_file_path) = &path_result {
+    if let Some(logging_file_path) = &logging_path_result {
+        // Ensure the logging directory is created
+        let parent_path = logging_file_path.parent().ok_or(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!(
+                "Unable to get the parent of the following Path: {}",
+                logging_file_path.display(),
+            ),
+        ))?;
+        fs::create_dir_all(parent_path)?;
+        // Attach it to fern
         fern_dispatch = fern_dispatch.chain(fern::log_file(logging_file_path)?);
-    }
+    };
 
     // Start fern
     fern_dispatch.apply()?;
+    info!("Logging with fern is setup");
 
-    // If the logging_file_path returned an error, print it with fern/logs
-    if let Err(e) = &path_result {
-        let message = "There was an issue getting the path for the log file";
-        warn!("{}: {:?}, falling back to stderr.", message, e);
+    // Log details of the logging_file_path result using fern/log
+    // Either logging the path fern is outputting to or the error from obtaining the path
+    match &logging_path_result {
+        Some(logging_file_path) => info!(
+            "Logging to the following file: {}",
+            logging_file_path.display()
+        ),
+        None => warn!("There was no path supplied for the log file, falling back to stderr."),
     }
-
-    info!("Logging with fern is setup");
     Ok(())
 }
 
+fn generate_logging_path(logfile_config: LogfileConfig) -> Result<PathBuf, io::Error> {
+    // Use the file name set in logfile_config or fallback to the default
+    let logfile_file_name = match logfile_config.file {
+        Some(file_name) => file_name,
+        None => PathBuf::from(XI_LOG_FILE),
+    };
```

Comment:
This could have a fallback for the case of an empty filename. Or disable file logging in that case? I guess that would be more useful.

Reply:
I'd probably lean towards the fallback & then have `--log-disable` or something as a flag for disabling logging.
But yea I do need to account for `" "`.